packages/ui/src/components/InputFields/PasswordInputField.tsx

```diff
@@ -0,0 +1,46 @@
+import type { ForwardedRef } from 'react';
+import { forwardRef, useRef } from 'react';
+import { useUpdateEffect, useToggle } from 'usehooks-ts';
+
+import PasswordHideIcon from '@/assets/icons/password-hide-icon.svg';
+import PasswordShowIcon from '@/assets/icons/password-show-icon.svg';
+import IconButton from '@/components/Button/IconButton';
+import useMergeRef from '@/hooks/use-merge-ref';
+
+import type { Props as InputFieldProps } from './InputField';
+import InputField from './InputField';
+
+type Props = Omit<InputFieldProps, 'type' | 'suffix' | 'isSuffixFocusVisible'>;
+
+const PasswordInputField = (props: Props, forwardRef: ForwardedRef<HTMLInputElement>) => {
+  const [showPassword, toggleShowPassword] = useToggle(false);
+  const innerRef = useRef<HTMLInputElement>(null);
+
+  // Merge inner ref with forwarded ref
+  const mergeRef = useMergeRef(forwardRef, innerRef);
+
+  // Refocus and move cursor to the end of the input field after password visibility is toggled
+  useUpdateEffect(() => {
+    if (innerRef.current) {
+      const { length } = innerRef.current.value;
+      innerRef.current.focus();
+      innerRef.current.setSelectionRange(length, length);
+    }
+  }, [showPassword]);
+
+  return (
+    <InputField
+      isSuffixFocusVisible
+      type={showPassword ? 'text' : 'password'}
+      suffix={
+        <IconButton onClick={toggleShowPassword}>
```

Comment:
You can set focus and selection range in the `onClick` handler.

Reply:
The idea was to keep the onClick event clean.